### PR TITLE
refactor(multiple): fix warnings related to division operator in latest version of Sass

### LINKED ISSUE #22866: bug(Material SCSS): [DEPRECATION WARNING] Using / for division is deprecated and will be removed in Dart Sass 2.0.0.
#### Reproduction

Steps to reproduce:
1. Upgrade angular 11 project to 12
2. Compile scss or run upgraded angular 12 project via serve command
 
#### Expected Behavior

/(divide) warning message should not come in console.

Related links - https://github.com/sass/sass/issues/2565

#### Actual Behavior

What behavior did you actually see?
image

#### Environment

- Angular: 12
- CDK/Material: "@angular/material": "^12.0.1",
- Browser(s): NA (Issue is coming while compiling scss or serving angular 12 project)
- Operating System (e.g. Windows, macOS, Ubuntu):  Windows

### PR DESCRIPTION
**Pull request by crisbeto**
The latest version of Sass prints a warning when the division operator is used. These changes migrate us to the recommended `math.div` function.

**Note:** there a few warnings left from MDC. I've submitted https://github.com/material-components/material-components-web/pull/7158 to address them.

More information: https://sass-lang.com/documentation/breaking-changes/slash-div

Fixes #22866.

### CODE REVIEW

File: src/material/core/typography/_typography-utils.scss [hunk 1]
```diff
@@ -78,7 +79,7 @@
     // Otherwise use the shorthand `font`, because it's the least amount of bytes. Note
     // that we need to use interpolation for `font-size/line-height` in order to prevent
     // Sass from dividing the two values.
-    font: $font-weight #{$font-size}/#{$line-height} $font-family;
+    font: $font-weight list.slash($font-size, $line-height) $font-family;
```
Comment (devversion) [inline src/material/core/typography/_typography-utils.scss:82]: I think the comment above needs to be updated accordingly.

**Review by devversion — APPROVED** [review]
LGTM, one minor nit from my side.

**Comment by crisbeto** [comment]
@jelbourn one thing we may want to consider here: the `math.div` function is available from Sass version 1.33.0 onwards whereas the CLI currently installs 1.32.12. Should we ask the CLI to update or delay this to a future version. Users that have updated Sass on their own will get a ton of warnings like in #22866. Alternatively, I can add our own helper for division that can be used to polyfill `math.div`.

**Review by andrewseguin — APPROVED** [review]

**Comment by crisbeto** [comment]
I've reworked the changes to make them backwards-compatible with Sass less than 1.34.0.

**Review by jelbourn — APPROVED** [review]
LGTM

**Review by devversion — APPROVED** [review]

**Comment by fredericojesus** [comment]
How do I know when this is going to be released ? Thanks in advance.

**Comment by crisbeto** [comment]
It'll be released in version 12.1.

**Comment by crhistianramirez** [comment]
I am seeing it in 12.0.6 but not in any of the newer versions like 12.1.1 for example. Is this expected?

**Comment by crisbeto** [comment]
That sounds weird since we only merged it into the 12.1.0 branch.

**Comment by crhistianramirez** [comment]
Never mind! My mistake 😅

**Pull request merged**